opendatahub-io/odh-dashboard
# Support for Trusty DB fields

https://issues.redhat.com/browse/RHOAIENG-13084

## Description


Refactored Trusty state as we are getting better status information.

Added DB modal. Screenshots below.

### Screenshots

![Screenshot 2024-10-07 at 1 08 29 PM](https://github.com/user-attachments/assets/990e94ce-c1e3-40f7-aad7-4c1462e6fb82)
![Screenshot 2024-10-07 at 1 14 30 PM](https://github.com/user-attachments/assets/82604b0b-eda1-49dc-ae17-c40e71006140)
![Screenshot 2024-10-07 at 1 14 40 PM](https://github.com/user-attachments/assets/b3efeb72-78fa-4412-999e-ed25bd608732)
![Screenshot 2024-10-07 at 12 51 24 PM](https://github.com/user-attachments/assets/f69be592-f1d9-4581-8b33-63b0bf400b25)


## How Has This Been Tested?



* Install Trusty in the DSC
    * ```
      trustyai:
        devFlags:
          manifests:
            - contextDir: config
              sourcePath: ''
              uri: https://api.github.com/repos/ruivieira/trustyai-service-operator/tarball/RHOAIENG-14041-db-status-dev
      ```
* Make sure our flag is enabled (may need to use the `?devFeatureFlags` param if you're on a cluster that is disabling trusty for RHOAI)

Deploy a DB & connect to it via the Trusty settings in the Project Settings tab.
Steps for mariadb: https://github.com/trustyai-explainability/reference/blob/main/db-connection/README.md

## Test Impact


Modal & button state tests -- dropped old checkbox tests (component was removed)

## Request review criteria:



Self checklist (all need to be checked):

If you have UI changes: 


After the PR is posted & before it merges:
